hal/src/electron/cellular_internal.cpp

```diff
@@ -12,8 +12,15 @@
 namespace detail {
 
 const int MCC_MNC_MIN_SIZE = 5;
+const int ICCID_MIN_SIZE = 19;
+const int ICCID_PREFIX_SIZE = 7;
 
-CellularNetProv _cellular_imsi_to_network_provider(const char* imsi) {
+CellularNetProv _cellular_sim_to_network_provider(const char* imsi, const char* iccid) {
+    if (iccid && strlen(iccid) >= ICCID_MIN_SIZE) {
+        if ((strncmp(iccid, "8988323",ICCID_PREFIX_SIZE) == 0) || (strncmp(iccid, "8988307",ICCID_PREFIX_SIZE) == 0)) {
```

Comment:
Nit: Magic numbers. The rest of the function is guilty of the same, but ideally these should be proper constants with appropriate names. Also, space after commas :)

Reply:
Used `const char` please verify if needed to assign these magic "numbers" differently.